app/src/main/java/com/muzima/controller/SmartCardController.java

```diff
@@ -0,0 +1,95 @@
+package com.muzima.controller;
+
+import android.util.Log;
+import com.muzima.api.model.SmartCardRecord;
+import com.muzima.api.service.SmartCardRecordService;
+import com.muzima.utils.StringUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.muzima.utils.Constants.DataSyncServiceConstants.SyncStatusConstants.UPLOAD_ERROR;
+
+public class SmartCardController {
+    public static final String TAG ="SmartCardController";
+
+    SmartCardRecordService smartCardRecordService;
+
+    public SmartCardController(SmartCardRecordService smartCardRecordService){
+        this.smartCardRecordService = smartCardRecordService;
+    }
+
+    public void saveSmartCardRecord(SmartCardRecord smartCardRecord) throws SmartCardRecordSaveException {
+        try {
+            smartCardRecordService.saveSmartCardRecord(smartCardRecord);
+        } catch (IOException e) {
+            throw new SmartCardRecordSaveException(e);
+        }
+    }
+
+    public void updateSmartCardRecord(SmartCardRecord smartCardRecord) throws SmartCardRecordSaveException {
+        try {
+            smartCardRecordService.updateSmartCardRecord(smartCardRecord);
+        } catch (IOException e) {
+            throw new SmartCardRecordSaveException(e);
+        }
+    }
+
+    public SmartCardRecord getSmartCardRecordByUuid(String uuid) throws SmartCardRecordFetchException {
+        try {
+            return smartCardRecordService.getSmartCardRecordByUuid(uuid);
+        } catch (IOException e) {
+            throw new SmartCardRecordFetchException(e);
+        }
+    }
+
+    public SmartCardRecord getSmartCardRecordByPersonUuid(String patientUuid) throws SmartCardRecordFetchException {
```

Comment:
Shouldn't the function be getSmartCardRecordByPatientUuid since we are passing patientUuid and not personUuid